docs(fix): Add "Suggest an Edit" link to "Make a Change Using the GitHub Web UI" page

aimeeu opened the pull request:
Using "Suggest and Edit" is much simpler than telling people how to find the page's source file in GitHub.

Renumbered numbered lists to use "1." for each bullet point.

aimeeu committed:
docs(fix): Add "Suggest an Edit" link

Using "Suggest and Edit" is much simpler than telling people how to
find the page's source file in GitHub.

dorbin commented on community/contributing/docs/github-changes.md:
```diff
@@ -9,10 +9,10 @@ sidebar:
-2. On the page where you see the issue, select the pencil icon at the top right. You can also scroll to the bottom of the page and select **Edit this page**.
-3. Make your changes in the GitHub markdown editor.
-4. Below the editor, fill in the **Propose file change** form.
+1. Click the **Suggest an Edit** link in the upper right hand corner of the Spinnaker documentation page. This takes you to the file in GitHub.
```
```suggestion
1. Click the **Suggest an Edit** link on the documentation page you want to update. This takes you to the file in GitHub.
```

dorbin replied:
The reason I suggest this change is that directional descriptions like "upper right" are considered less accessible. Also, if we move the link this'll be broken. But unless we change the text of the link, this will still be findable.

dorbin commented on community/contributing/docs/github-changes.md:
```diff
@@ -9,10 +9,10 @@ sidebar:
-3. Make your changes in the GitHub markdown editor.
-4. Below the editor, fill in the **Propose file change** form.
+1. Click the **Suggest an Edit** link in the upper right hand corner of the Spinnaker documentation page. This takes you to the file in GitHub.
+1. Select the pencil icon at the top right. You can also scroll to the bottom of the page and select **Edit this page**.
```
Same thing here. Maybe "pencil icon at the top right" --> "**edit** button" or "**Edit this file** button".

aimeeu replied:
I followed the steps. I didn't see an "Edit this page" link or button anywhere, only the pencil icon. Removed the directional but left pencil icon.

dorbin replied:
"Edit this file" is the hover text for the icon, so that's what someone using a sceen reader would hear. Of course, that confuses sighted users, who only see the pencil. :) No biggie for now, but I'll try to remember to look up best practices for this.

dorbin replied:
Something like "Click the **Edit this file** (✏️) button."

aimeeu replied:
where did you get that pencil icon? Unfortunately I can't see the comment source.

aimeeu replied:
Added an image that points out the pencil icon.

dorbin commented on community/contributing/docs/github-changes.md:
```diff
@@ -9,10 +9,10 @@ sidebar:
+1. Click the **Suggest an Edit** link in the upper right hand corner of the Spinnaker documentation page. This takes you to the file in GitHub.
+1. Select the pencil icon at the top right. You can also scroll to the bottom of the page and select **Edit this page**.
+1. Make your changes in the GitHub markdown editor.
+1. Below the editor, fill in the **Propose file change** form.
```
Also might want to avoid this "Below" directionality.

dorbin commented on community/contributing/docs/github-changes.md:
```diff
@@ -28,13 +28,13 @@ sidebar:
   For more information, see [commit message guidelines](https://www.spinnaker.io/community/contributing/submitting/#commit-message-conventions).
 
-  In the **second field**, provide a clear and detailed description of your PR. Do not to leave these feilds blank, as it is helpful to reviewers when merging your request to have additional context as to what your pull request is about.
+  In the **second field**, provide a clear and detailed description of your PR. Do not to leave these fields blank, as it is helpful to reviewers when merging your request to have additional context as to what your pull request is about.
```
These "first field," "second field" don't work to well either, but that's how it was before this PR, so ok for now. (Heck, I probably wrote that myself 🌵 )

aimeeu replied:
@dorbin A new contributor submitted this PR during Spinnaker Gardening Days. After reading your comments, I followed the instructions, and then reordered. Also ensured field names matched the UI. Moved the PR title checker content to the `Open a pull request` step because GitHub only checks the PR title, not the title of the commit. Added two images for clarity.

dorbin replied:
👍

aimeeu committed:
Clarify that clicking the link takes you to page's source file.

aimeeu committed:
Reorder steps and add screenshots.

Move PR title checker info to the Open a pull request section.
Add images.
Ensure UI fields capitalization matches the UI.

aimeeu committed:
Add image files.

aimeeu committed:
Add image for pencil icon

dorbin committed:
Merge branch 'master' into updateGitHubPage

dorbin commented on community/contributing/docs/github-changes.md:
```diff
@@ -10,7 +10,10 @@ sidebar:
 
 1. Click the **Suggest an Edit** link on the documentation page you want to update. This takes you to the page's source file in GitHub.
-1. Click the pencil icon to edit the file.
+1. Click the **Edit this file in the fork of your project** pencil icon to edit the file.
```
Hmm, for me it just says "Edit this file"

![edit_this_file](https://user-images.githubusercontent.com/29312809/80819669-6c3ba600-8ba3-11ea-88ef-b76e76fbda01.png)

dorbin replied:
Either way, I think it's fine just to say "**Edit this file** icon"

aimeeu replied:
That's bizarre. I'm on a Mac, and the hover help is the same regardless of browser.  I don't see the computer monitor icon in your screenshot either.
![edit-the-file](https://user-images.githubusercontent.com/26799583/80821482-6db69f80-8b9e-11ea-83d0-cfd9ca7d01f7.png)

aimeeu replied:
Changed to "Edit this file"

dorbin replied:
@aimeeu do you have merge rights in the repo? It's possible that I'm seeing something different because I do. Thus you're seeing what the user will see, and I'm not. "Edit this file..." with the elipsis, seems like the safe choice.

aimeeu replied:
Nope - no merge rights. I'm just a lowly contributor. LOL. 
That explains why you don't see the "in your fork" portion.

aimeeu committed:
Replace "Edit this file..." with "Edit this file"

aimeeu committed:
Change to "Edit this file..."
